corehq/apps/users/views/mobile/users.py

```diff
@@ -1226,6 +1241,43 @@ def _delete_users(self, request, user_docs_by_id, user_ids_with_forms):
             messages.success(request, f"{deleted_count} user(s) deleted.")
 
 
+class CommCareUsersLookup(BaseManageCommCareUserView, UsernameUploadMixin):
+    urlname = 'commcare_users_lookup'
+    page_title = ugettext_noop('Mobile Workers Bulk Lookup')
+    template_name = 'users/bulk_lookup.html'
+
+    @property
+    def page_context(self):
+        context = self.main_context
+        context.update({
+            'bulk_upload_form': get_bulk_upload_form(),
+        })
+        return context
+
+    def post(self, request, *args, **kwargs):
+        outfile = io.BytesIO()
+        writer = Excel2007ExportWriter()
+        tab_name = "users"
+        header_table = [(tab_name, [(_("username"), _("exists"))])]
+        writer.open(header_table=header_table, file=outfile)
+        usernames = self._get_usernames(request)
+        if not usernames:
+            return self.get(request, *args, **kwargs)
+
+        user_docs_by_id = {doc['_id']: doc for doc in get_user_docs_by_username(usernames)}
+        usernames_not_found = self._get_usernames_not_found(request, user_docs_by_id, usernames, strict=False)
+
+        rows = []
+        for username in usernames:
+            rows.append([raw_username(username), _("no") if username in usernames_not_found else _("yes")])
+        writer.write([(tab_name, rows)])
```

Comment:
nit: I think we can extract this logic for creating the excel sheet in a different method.

Reply:
c644b441c75